src/deepsparse/transformers/pipelines/text_generation.py

```diff
@@ -877,39 +908,46 @@ def join_engine_outputs(
         :param orig_batch_size: The original batch size
         :return: A list of joined outputs
         """
-        tokens, logits, finish_reason = zip(*batch_outputs)
-        if self.cache_support_enabled:
-            # if the model has kv cache, we need to account for
-            # the fact that the predicted outputs may have
-            # different lengths
-
-            # find the longest sequence in the batch of tokens
-            max_len = max([token.shape[1] for token in tokens])
-
-            # pad all tokens to the same length
-            tokens = [
-                pad_to_fixed_length(
-                    array=prediction,
-                    max_len=max_len,
-                    value=self.tokenizer.pad_token_id,
-                    axis=1,
-                )
-                for prediction in tokens
-            ]
+        streaming = kwargs.get("streaming")
+        if streaming:
+            for batch in batch_outputs:
+                for outputs in batch:
+                    yield outputs
+        else:
+            batch_outputs = [list(*b) for b in batch_outputs]
+            tokens, logits, finish_reason = zip(*batch_outputs)
+            if self.cache_support_enabled:
+                # if the model has kv cache, we need to account for
+                # the fact that the predicted outputs may have
+                # different lengths
+
+                # find the longest sequence in the batch of tokens
+                max_len = max([token.shape[1] for token in tokens])
+
+                # pad all tokens to the same length
+                tokens = [
+                    pad_to_fixed_length(
+                        array=prediction,
+                        max_len=max_len,
+                        value=self.tokenizer.pad_token_id,
+                        axis=1,
+                    )
+                    for prediction in tokens
+                ]
 
-            # find the longest sequence in the batch of logits
-            max_len = max([logits.shape[1] for logits in logits])
+                # find the longest sequence in the batch of logits
+                max_len = max([logits.shape[1] for logits in logits])
```

Comment:
```suggestion
                max_len = max(logits.shape[1] for logits in logits)
```

Reply:
gonna leave these for now since unrelated to the current PR